### DISCUSSION TITLE
Reconhecer QRCode e Código de Barras de imagem

### DISCUSSION THREAD

**Discussion by pcomora** [Flutterando/forum#94] [Perguntas e Respostas]
Bom Dia

Estou iniciando em Flutter e gostaria de saber se tem como através de imagem identificar QRCode e Código de Barras?

Preciso passar uma imagem através da camera do aparelho ou uma imagem da galeria, fazer a leitura da imagem e identificar de nesta imagem existe um QRCode e ler esse QRCode. Eu já consegui fazer isso em python, mas não se há como fazer usando dart.

Desde já obrigado.
Paulo Comora

**Comment by eduardoflorence** [comment]
Olá @pcomora,
    No Flutter você pode utilizar o Google ML Kit.
    O package é esse: https://pub.dev/packages/google_ml_kit
    Mais especificamente, você vai utilizar o [Barcode Scanning](https://developers.google.com/ml-kit/vision/barcode-scanning), que apesar do nome, consegue lê vários formatos de código de barras e também QR Code.
    Outra alternativa é utilizar o package https://pub.dev/packages/mobile_scanner, que utiliza o Google ML Kit, mas facilita o trabalho.

**Reply by pcomora** [reply]
Obrigado, vou seguir a recomendação.
